src/main/java/com/untamedears/ItemExchange/utility/ExchangeRule.java

```diff
@@ -545,47 +548,79 @@ public int checkMultiples(Inventory inventory) {
 	 * Checks if the given ItemStack follows the ItemRules except for the amount
 	 */
 	public boolean followsRules(ItemStack itemStack) {
-		// check material type and druability
-		boolean followsRules = DeprecatedMethods.getMaterialId(material) == DeprecatedMethods.getItemId(itemStack)
-				&& durability == itemStack.getDurability();
-		
-		// Check enchantments
-		if (itemStack.getEnchantments().size() > 0) {
-			followsRules = followsRules && itemStack.getEnchantments().entrySet().containsAll(requiredEnchantments.entrySet());
-			for (Enchantment excludedEnchantment : excludedEnchantments) {
-				followsRules = followsRules && !itemStack.getEnchantments().entrySet().contains(excludedEnchantment);
+		// If not the same material, return false
+		if (DeprecatedMethods.getMaterialId(material) != DeprecatedMethods.getItemId(itemStack)) {
+			return false;
+		}
+		// If not the same durability, return false
+		if (durability != itemStack.getDurability()) {
+			return false;
+		}
+		// Check required enchantments
+		Map<Enchantment, Integer> itemEnchants = itemStack.getEnchantments();
+		if (itemEnchants.isEmpty() != requiredEnchantments.isEmpty()) {
```

Comment:
Everything from here to line 574 could be simplified to something like:

```
if (!unlistedEnchantmentsAllowed && itemEnchants.size() != requiredEnchantments.size()) {
				return false;
}
if (!itemEnchants.entrySet().containsAll(requiredEnchantments.entrySet())) {
				return false;
}

```

Reply:
I wasn't sure if comparing two empty collections in a .containsAll() would return true or false. It turns out that an empty collection _does_ contain all the elements from another empty collection. Huh, TIL. But it means I can do the simplification you suggested